### DISCUSSION TITLE
Long text increment cells height in the reduced table

### DISCUSSION THREAD

**Discussion by raquelarrojo** [dxc-technology/halstack-react#1946] [General] [answered]
When there is long text in the cells of a `reduced` table, there is a break line and visually, it is similar to the `default` table.

In this snapshot, all the tables are reduced tables, but Bookmarks and Tasks tables are visually like a default table due to the long text in their cells. 

![image](https://github.com/dxc-technology/halstack-react/assets/10975289/90575bdc-cfdb-425d-ac9d-befdcb61e848)

Should we add ellipsis in the cells or any other way so that the cell height does not change?

@lcordobapuerto @rconcepcion3 @irenevera @thiyaDezine

**Comment by irenevera** [comment]
If we indicate that the text has been truncated using ellipses will we provide the full text somehow? (ie: tooltip when the mouse hovers the cell)

**Comment by rconcepcion3** [comment]
Hmmm... the reason we're implementing reduced size tables in the first place is to be able to show more rows of content. 

In this case, my opinion is yes, keeping the text to one line then adding an ellipsis makes sense.

I agree with Irene's input... we truncate and add the ellipsis but we should show the full text within a tooltip on hovering over the truncated text... 

This might be a good starting point for a hover/tooltip component discussion :)

https://carbondesignsystem.com/components/tooltip/usage/

**Comment by ThiyaDezine** [comment]
Yes, Character length must be standardized in order to use ellipsis. 

An additional choice would be able to expand and collapse the row to see the complete description.
![Screenshot (847)](https://github.com/dxc-technology/halstack-react/assets/122968305/c4a0bc00-1e0e-4ba4-9c10-c27c98e46b98)

**Reply by kukoc** [reply]
I like both enhancements. Adding the ellipsis would be easier at this stage compared to make the column with dynamic. We should start by finishing the tooltip opinion and implementation and then use it in our table. In my opinion we should add these items to the backlog and work on them in priority since it's a limitation that is impacting the UX of a project.

**Comment by irenevera** [comment]
Just thinking out loud here:
- If we go for the tooltip solution, that new component can be used in many other places and maybe is easier to implement (quick win?)
- If we truncate text can user still copy the whole text?
- While truncating + tooltip is good solution (specially to reduce visual clutter), can happen than relevant information is hidden and cannot be easily compared between cells. Shall we specify the ideal usage of reduced size tables (for short legnth text)?
- While I really love Thiagu's input, I believe the usage is slightly different than the problem we are trying to solve here. Expanded rows are normally used to show additional content within a row entry.

**Comment by lcordobapuerto** [comment, marked as answer]
After discussing this matter with the team, we have decided that the best solution is to truncate the text with an ellipsis. In parallel, we are working on a new Tooltip component from which the full text can be displayed and copied if necessary. 
Thank you all for your valuable inputs!
